### PR TITLE
http: Added a Gin http receiver sample

### PR DESCRIPTION
**Pull request by dgeorgievski**
A simple example of a Gin http receiver for CloudEvents. The  Gorilla/mux  project is archived which makes the existing gorilla receiver example deprecated.

### CODE REVIEW

**Comment by duglin** [comment]
@dgeorgievski can you rebase your PR?

**Comment by dgeorgievski** [comment]
Rebased

**Comment by duglin** [comment]
thanks!
LGTM
ping @lionelvillard @embano1

File: samples/http/receiver-gin/README.md [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+
+An example of a Gin webframework CloudEvents receiver with a [TektonEvent](https://tekton.dev/docs/pipelines/events/)
+
+# Steps
+
+Get dependencies
+```shell
+cd samples/
+ go get github.com/gin-gonic/gin
+ go get github.com/rs/zerolog/log
```
Comment (embano1) [inline samples/http/receiver-gin/README.md:12]:
```suggestion
 go get .
```

File: samples/http/receiver-gin/README.md [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+
+```
+
+Run the app
+```shell
+ go run main.go
+```
+
+Test a CloudEvent
```
Comment (embano1) [inline samples/http/receiver-gin/README.md:21]:
```suggestion
Send a CloudEvent
```

File: samples/http/receiver-gin/README.md [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+```shell
+curl -v \
+ -H "Ce-Id: e7d95c20-6eb4-4614-946d-27b0ce41c7ff" \
+ -H "Ce-Source: /apis///namespaces/dimitar//clone-build-n4qhgl" \
```
Comment (embano1) [inline samples/http/receiver-gin/README.md:25]:
```suggestion
 -H "Ce-Source: /apis/namespaces/dimitar/clone-build-n4qhgl" \
```

Comment (embano1) [reply samples/http/receiver-gin/README.md:25]: or is the above a real value from the example event?

File: samples/http/receiver-gin/main.go [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+func index(c *gin.Context) {
+	c.JSON(http.StatusOK, "Welcome to CloudEvents")
+}
+
+func healthz(c *gin.Context) {
+	c.String(http.StatusOK, "OK")
+}
+
+func cloudEventsHandler() gin.HandlerFunc {
+
```
Comment (embano1) [inline samples/http/receiver-gin/main.go:26]: nit: remove

File: samples/http/receiver-gin/main.go [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+				Err(err).
+				Msg("failed to create handler")
+		}
+
+		ceh.ServeHTTP(c.Writer, c.Request)
+	}
+}
+
+func main() {
+
```
Comment (embano1) [inline samples/http/receiver-gin/main.go:47]: nit: remove

**Review by embano1 — COMMENTED** [review]
Minor comments

**Pull request merged**